src/backend/optimizer/plan/createplan.c

```diff
@@ -3768,8 +3768,7 @@ create_hashjoin_plan(PlannerInfo *root,
 	 * (allowing us to check the outer for rows before building the
 	 * hash-table).
 	 */
-	if (best_path->jpath.outerjoinpath == NULL ||
-		best_path->jpath.outerjoinpath->motionHazard ||
+	if (best_path->jpath.outerjoinpath->motionHazard ||
```

Comment:
Please update description with more details about how create_hashjoin_path() is involved. Can you provide context about this comment above?
https://github.com/arenadata/gpdb/blob/d1b3eac27b79bcb776fdddac624c16d8d19270ab/src/backend/optimizer/plan/createplan.c#L3757-L3760

Reply:
Here the `best_path->jpath.outerjoinpath` pointer cannot be NULL because it is dereferenced [above](https://github.com/arenadata/gpdb/pull/725/commits/d1b3eac27b79bcb776fdddac624c16d8d19270ab#diff-4f0575423ee0e1a3434c5af37bdc873165983c0884489e5ae2c71250c66a0ec3R3691).

Reply:
Can you add an assert to above the `get_switched_clauses()` call?

Reply:
added